### PR TITLE
Extensions: WPJM - Save "Job Submission" settings

### PR DESCRIPTION
**Pull request by donnapep**
This PR saves the settings on the _Job Submission_ tab. It also makes a small tweak to the UI such that the _Role_ dropdown is only visible if the _Enable account creation during submission_ setting is enabled.

## Testing

Verify each of the different sections save their settings independently, and that the _Role_ dropdown is only visible if the _Enable account creation during submission_ setting is enabled.

### CODE REVIEW

File: client/extensions/wp-job-manager/components/settings/job-submission/index.jsx [hunk 1]
```diff
@@ -21,185 +21,208 @@ import ReduxFormTextInput from 'components/redux-forms/redux-form-text-input';
...
+									{ translate( 'Users can continue to edit pending listings until they are approved by an admin.' ) }
+								</FormSettingExplanation>
+							</FormFieldset>
+						</Card>
+					</FormSection>
+				</form>
+
+				<form>
+					<FormSection name="duration">
+						<SectionHeader label={ translate( 'Listing Duration' ) }>
```
Comment (a8ci18n) [inline client/extensions/wp-job-manager/components/settings/job-submission/index.jsx]: Hi! I've found a possible matching string that has already been translated **32** times:
`translate( 'Duration' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/wp-job-manager/components/settings/job-submission/index.jsx [hunk 2]
```diff
@@ -21,185 +21,208 @@ import ReduxFormTextInput from 'components/redux-forms/redux-form-text-input';
...
+										'your own method of assigning roles will apply.' ) }
+								</FormSettingExplanation>
+							</FormFieldset>
+						</Card>
+					</FormSection>
+				</form>
+
+				<form>
+					<FormSection name="approval">
+						<SectionHeader label={ translate( 'Approval' ) }>
```
Comment (a8ci18n) [inline client/extensions/wp-job-manager/components/settings/job-submission/index.jsx]: Hi! I've found a possible matching string that has already been translated **3** times:
`translate( 'Waiting approval' )` *ES Score: 9*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/wpjm-save-job-submission-settings)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/wp-job-manager/components/settings/job-submission/index.jsx [hunk 3]
```diff
@@ -21,185 +21,208 @@ import ReduxFormTextInput from 'components/redux-forms/redux-form-text-input';
...
-	translate: PropTypes.func,
-};
+									name="role">
+									<option value="editor">{ translate( 'Editor' ) }</option>
+									<option value="author">{ translate( 'Author' ) }</option>
+									<option value="contributor">{ translate( 'Contributor' ) }</option>
+									<option value="subscriber">{ translate( 'Subscriber' ) }</option>
+									<option value="teacher">{ translate( 'Teacher' ) }</option>
+									<option value="employer">{ translate( 'Employer' ) }</option>
+									<option value="customer">{ translate( 'Customer' ) }</option>
```
Comment (a8ci18n) [inline client/extensions/wp-job-manager/components/settings/job-submission/index.jsx]: Hi! I've found a possible matching string that has already been translated **35** times:
`translate( 'Customer Name' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/wp-job-manager/components/settings/job-submission/index.jsx [hunk 4]
```diff
@@ -21,191 +21,219 @@ import ReduxFormTextInput from 'components/redux-forms/redux-form-text-input';
...
+									<ReduxFormSelect
+										disabled={ isDisabled }
+										name="role">
+										<option value="editor">{ translate( 'Editor' ) }</option>
+										<option value="author">{ translate( 'Author' ) }</option>
+										<option value="contributor">{ translate( 'Contributor' ) }</option>
+										<option value="subscriber">{ translate( 'Subscriber' ) }</option>
+										<option value="teacher">{ translate( 'Teacher' ) }</option>
+										<option value="employer">{ translate( 'Employer' ) }</option>
+										<option value="customer">{ translate( 'Customer' ) }</option>
```
Comment (a8ci18n) [inline client/extensions/wp-job-manager/components/settings/job-submission/index.jsx]: Hi! I've found a possible matching string that has already been translated **35** times:
`translate( 'Customer Name' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by ockham** [comment]
> It also makes a small tweak to the UI such that the _Role_ dropdown is only visible if the _Enable account creation during submission_ setting is enabled.

Should we move the toggle one position below so the dropdown is right below?

**Comment by ockham** [comment]
>> It also makes a small tweak to the UI such that the _Role_ dropdown is only visible if the _Enable account creation during submission_ setting is enabled.

> Should we move the toggle one position below so the dropdown is right below?

cc @donnchawp since wp-admin should probably align with this change.

**Pull request merged**